build/blog.ts

```diff
@@ -42,12 +46,24 @@ function calculateReadTime(copy: string): number {
 }
 
 async function readPost(
-  file: string
+  file: string,
+  extraMeta = true
 ): Promise<{ blogMeta: BlogPostFrontmatter; body: string }> {
   const raw = await fs.readFile(file, "utf-8");
 
   const { attributes, body } = frontmatter<BlogPostFrontmatter>(raw);
```

Comment:
🟠 Can we make these three lines a separate - memoized - function `readRawPost()`? That one could be used in `allPostFrontmatter()`, which would make the `extraMeta` property superfluous.

Reply:
We still need the read time in `allPostFrontmatter()`, and to parse the rest of the frontmatter, so it's not really reading the raw post - I'll keep the options object

Reply:
These are the three lines I referred to:

```ts
  const raw = await fs.readFile(file, "utf-8");

  const { attributes, body } = frontmatter<BlogPostFrontmatter>(raw);
```

I'd expect these to take most of the build time. Calculating the read time should be cheap in comparison.